api/src/main/java/org/pmiops/workbench/api/SurveysController.java

```diff
@@ -0,0 +1,48 @@
+package org.pmiops.workbench.api;
+
+import javax.inject.Provider;
+import org.pmiops.workbench.db.dao.NewUserSatisfactionSurveyDao;
+import org.pmiops.workbench.db.model.DbNewUserSatisfactionSurvey;
+import org.pmiops.workbench.db.model.DbUser;
+import org.pmiops.workbench.exceptions.BadRequestException;
+import org.pmiops.workbench.model.CreateNewUserSatisfactionSurvey;
+import org.pmiops.workbench.survey.NewUserSatisfactionSurveyMapper;
+import org.pmiops.workbench.survey.NewUserSatisfactionSurveyService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class SurveysController implements SurveysApiDelegate {
+  private final NewUserSatisfactionSurveyDao newUserSatisfactionSurveyDao;
+  private final Provider<DbUser> userProvider;
+  private final NewUserSatisfactionSurveyMapper newUserSatisfactionSurveyMapper;
+  private final NewUserSatisfactionSurveyService newUserSatisfactionSurveyService;
+
+  @Autowired
+  SurveysController(
+      NewUserSatisfactionSurveyDao newUserSatisfactionSurveyDao,
+      Provider<DbUser> userProvider,
+      NewUserSatisfactionSurveyMapper newUserSatisfactionSurveyMapper,
+      NewUserSatisfactionSurveyService newUserSatisfactionSurveyService) {
+    this.newUserSatisfactionSurveyDao = newUserSatisfactionSurveyDao;
+    this.userProvider = userProvider;
+    this.newUserSatisfactionSurveyMapper = newUserSatisfactionSurveyMapper;
+    this.newUserSatisfactionSurveyService = newUserSatisfactionSurveyService;
+  }
+
+  @Override
+  public ResponseEntity<Void> createNewUserSatisfactionSurvey(
+      CreateNewUserSatisfactionSurvey createNewUserSatisfactionSurvey) {
+    DbUser user = userProvider.get();
+    if (!newUserSatisfactionSurveyService.eligibleToTakeSurvey(user)) {
+      throw new BadRequestException("User is ineligible to take this survey.");
+    }
+    DbNewUserSatisfactionSurvey newUserSatisfactionSurvey =
+        newUserSatisfactionSurveyMapper.toDbNewUserSatisfactionSurvey(
+            createNewUserSatisfactionSurvey, user);
+    newUserSatisfactionSurveyDao.save(newUserSatisfactionSurvey);
```

Comment:
We have an unusual pattern throughout the codebase when saving DB entity objects, like `foo = fooDao.save(foo);`.  I don't fully understand the reason why it's necessary, but if we skip the "save to yourself" step and then try to reuse the DB entity object, Hibernate often has strange behavior.

It's not relevant here because you're not using `newUserSatisfactionSurvey` again, but it's a good habit to get into.

Reply:
Thanks, this is good to know. Updated. I suspect Hibernate objects cache DB calls and isn't smart about refreshing them so we need to manually refresh them.